network/network.go

```diff
@@ -253,14 +255,50 @@ func (n *Network) Start() error {
 		prot.Start()
 	}
 
+	return n.connectToKnownNodes(nodeDID)
+}
+
+func (n *Network) connectToKnownNodes(nodeDID did.DID) error {
 	// Start connecting to bootstrap nodes
 	for _, bootstrapNode := range n.config.BootstrapNodes {
 		if len(strings.TrimSpace(bootstrapNode)) == 0 {
 			continue
 		}
-		n.connectionManager.Connect(bootstrapNode, transport.WithUnauthenticated())
+		n.connectionManager.Connect(transport.Address(bootstrapNode), transport.WithUnauthenticated())
+	}
+
+	if !n.config.EnableDiscovery {
+		return nil
 	}
 
+	// start connecting to published NutsComm addresses
+	otherNodes, err := n.didDocumentFinder.Find(doc.IsActive(), doc.ValidAt(time.Now()), doc.ByServiceType(transport.NutsCommServiceType))
+	if err != nil {
+		return err
+	}
+	for _, node := range otherNodes {
+		if !nodeDID.Empty() && node.ID.Equals(nodeDID) {
+			// Found local node, do not discover.
+			continue
+		}
+	inner:
+		for _, service := range node.Service {
+			if service.Type == transport.NutsCommServiceType {
+				var nutsCommStr string
+				if err = service.UnmarshalServiceEndpoint(&nutsCommStr); err != nil {
+					log.Logger().Warnf("failed to extract NutsComm address from service (did=%s): %v", node.ID.String(), err)
+					continue inner
+				}
+				address, err := transport.ParseAddress(nutsCommStr)
+				if err != nil {
+					log.Logger().Warnf("invalid NutsComm address from service (did=%s, str=%s): %v", node.ID.String(), nutsCommStr, err)
+					continue inner
+				}
+				log.Logger().Infof("Discovered Nuts node (address=%s), published by %s", address, node.ID)
+				n.connectionManager.Connect(address)
+			}
+		}
+	}
```

Comment:
Is this only called once at startup?

Reply:
That was the initial thought. The best way to adjust things runtime is to receive an event in the network layer from the VDR.